### PR TITLE
[Doc] CTCLoss: clarify that log_probs must come from log_softmax

### LINKED ISSUE #181534: CTCLoss backward is inconsistent with its forward derivative with respect to the documented log_probs input.
### 🐛 Describe the bug

```py
import torch
import torch.nn as nn
import torch.nn.functional as F

torch.manual_seed(123)

T, N, C = 3, 1, 3  # time, batch, classes incl. blank
log_probs = F.log_softmax(torch.randn(T, N, C), dim=-1).double()
log_probs.requires_grad_(True)

targets = torch.tensor([[1]])
input_lens = torch.tensor([T])
target_lens = torch.tensor([1])

loss_fn = nn.CTCLoss(blank=0)

def f(x):
    return loss_fn(x, targets, input_lens, target_lens)

print(torch.autograd.gradcheck(f, (log_probs,), raise_exception=True))
```

Environment:
- PyTorch: 2.11.0+cu128
- Python: 3.13.5

The following minimal example reproduces the issue. `gradcheck` fails with a clear Jacobian mismatch for the input `log_probs`.

The numerical Jacobian contains exact zeros for class 2 at every time step:

  numerical:
  tensor([[-0.4727],
          [-0.5273],
          [ 0.0000],
          [-0.5624],
          [-0.4376],
          [ 0.0000],
          [-0.4784],
          [-0.5216],
          [ 0.0000]], dtype=torch.float64)

However, the analytical gradient returned by CTCLoss backward gives non-zero values at the same positions:

  analytical:
  tensor([[-0.1430],
          [-0.1116],
          [ 0.2547],
          [-0.2236],
          [-0.3075],
          [ 0.5311],
          [-0.3088],
          [-0.3109],
          [ 0.6198]], dtype=torch.float64)

Since class 2 is neither the blank class nor the target label in this example, the true derivative with respect to `log_probs[..., 2]` should be zero. The numerical Jacobian confirms this, but the analytical backward returns non-zero gradients.

This suggests that `CTCLoss.backward()` is inconsistent with the derivative of `CTCLoss.forward()` with respect to the documented `log_probs` input.

### Versions

PyTorch: 2.11.0+cu128
Python: 3.13.5

cc @ezyang @gchanan @kadeng @msaroufim @albanD @gqchen @nikitaved @soulitzer @Varal7 @bobrenjc93 @mruberry @jbschlosser @walterddr @mikaylagawarecki

### PR DESCRIPTION
**Pull request by Arindamsamanta2004**
## Summary

The CTCLoss docs currently state that `log_probs` should be log-probabilities "e.g. obtained with `log_softmax`", which makes `log_softmax` sound optional. However, the backward pass computes gradients with respect to the pre-softmax
logits, so `log_softmax` must be part of the autograd graph for gradients to be correct. Without it, the forward loss is correct but the gradients are silently incorrect.

This PR updates the documentation to make this requirement explicit by changing
"e.g." to "must" and adding a note explaining the backward behavior.

Fixes #181534  
Related: #82545, #52241

## Test plan

-  gradcheck passes with log_softmax in the graph, fails without (expected)
- lintrunner clean on both files

cc @svekars @sekyondaMeta @AlannaBurke @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

### CODE REVIEW

**Comment by Arindamsamanta2004** [comment]
Hi @mikaylagawarecki , can you help me with the review ? Thanks .
